src/go/rdctl/cmd/root.go

```diff
@@ -61,101 +62,120 @@ func Execute() {
+	rootCmd.PersistentFlags().StringVar(&host, "host", "", "default is localhost; most useful for WSL")
+	rootCmd.PersistentFlags().StringVar(&port, "port", "", "overrides the port setting in the config file")
+	rootCmd.PersistentFlags().StringVar(&password, "password", "", "overrides the password setting in the config file")
 }
 
-func doRequest(method string, command string)  error {
-  req, err := getRequestObject(method, command)
-  if err != nil {
-    return err
-  }
-  return doRestOfRequest(req)
+func doRequest(method string, command string) ([]byte, error) {
+	req, err := getRequestObject(method, command)
+	if err != nil {
+		return nil, err
+	}
+	return doRestOfRequest(req)
+}
+
+func doRequestWithPayload(method string, command string, payload *bytes.Buffer) ([]byte, error) {
+	req, err := http.NewRequest(method, fmt.Sprintf("http://%s:%s/%s/%s", host, port, apiVersion, command), payload)
+	// TODO: Delete this or add a --debug flag: fmt.Fprintf(os.Stderr, `method: %s, URL: "http://%s:%s/%s/%s", payload: %s\n`, method, host, port, apiVersion, command, payload.String())
+	if err != nil {
+		return nil, err
+	}
+	req.SetBasicAuth(user, password)
+	req.Header.Add("Content-Type", "application/json")
+	req.Close = true
+	return doRestOfRequest(req)
 }
 
 func getRequestObject(method string, command string) (*http.Request, error) {
-  req, err := http.NewRequest(method, fmt.Sprintf("http://%s:%s/%s/%s", host, port, apiVersion, command), nil)
-  if err != nil {
-    return nil, err
-  }
-  req.SetBasicAuth(user, password)
-  req.Header.Add("Content-Type", "text/plain")
-  req.Close = true
-  return req, nil
+	req, err := http.NewRequest(method, fmt.Sprintf("http://%s:%s/%s/%s", host, port, apiVersion, command), nil)
+	if err != nil {
+		return nil, err
+	}
+	req.SetBasicAuth(user, password)
+	req.Header.Add("Content-Type", "text/plain")
+	req.Close = true
+	return req, nil
 }
 
-func doRestOfRequest(req *http.Request) error {
-  client := http.Client{}
-  response, err := client.Do(req)
-  if err != nil {
-    return err
-  }
-  if response.StatusCode < 200 || response.StatusCode >= 300 {
-    switch (response.StatusCode) {
-    case 401:
-      return fmt.Errorf("user/password not accepted")
-    case 500:
-      return fmt.Errorf("server-side problem: please consult the server logs for more information.")
-    }
-    return fmt.Errorf("%s", response.Status)
-  }
-
-  defer response.Body.Close()
-
-  body, err := ioutil.ReadAll(response.Body)
-  if err != nil {
-    return err
-  }
-
-  fmt.Println(string(body))
-  return nil
+func doRestOfRequest(req *http.Request) ([]byte, error) {
+	client := http.Client{}
+	response, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	statusMessage := ""
+	if response.StatusCode < 200 || response.StatusCode >= 300 {
+		switch response.StatusCode {
+		case 400:
+			// rest of message should be in the body
+			statusMessage = "error in request"
```

Comment:
This is never used, because there's a `return` after the `switch` (inside the current `if`).

Reply:
Fixed it.